buildlib/tools/builds.sh

```diff
@@ -191,21 +191,26 @@ build_prof() {
 # Build UGNI
 #
 build_ugni() {
-	echo "==== Build with cray-ugni ===="
-	#
-	# Point pkg-config to contrib/cray-ugni-mock, and replace
-	# PKG_CONFIG_TOP_BUILD_DIR with source dir, since the mock .pc files contain
-	# relative paths.
-	#
-	${WORKSPACE}/contrib/configure-devel --prefix=$ucx_inst --with-ugni \
-		PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/${WORKSPACE}/contrib/cray-ugni-mock \
-		PKG_CONFIG_TOP_BUILD_DIR=${WORKSPACE}
-	$MAKEP
-
-	# make sure UGNI transport is enabled
-	grep '#define HAVE_TL_UGNI 1' config.h
-
-	$MAKEP distcheck
+        # Disable building ugni for the rocm docker image.
+        if [ -f /opt/rocm/bin/rocminfo ]; then
+                  azure_log_warning "Not building with cray-ugni"
```

Comment:
why?

Reply:
the ugni stuff doesn't compile on the rocm image (its missing xpmem), I wanted to find a way to exclude it from compiling it with the rocm image.

Reply:
is there a better way for detecting whether ugni can be compiled? Many of the other functions rely on a module load succeeding or not, but there is nothing comparable for ugni

Reply:
> the ugni stuff doesn't compile on the rocm image (its missing xpmem)

Can you pls post the error?

Reply:
yes, give me a min. I noticed that configure succeeds with the ugni option when trying to run it separately in the same docker image, so I will try to re-trigger the CI build to see whether it still fails (and remove at least temporarily the change in the script).

Reply:
well, it looks like its passing now in the CI as well, not entirely sure what change triggered the test to pass/fail. But I was able to undo the change to the build_ugni() function and it passes that test, so we should be good.

Reply:
actually, I know what the problem was. I misread the configure error early on, I though it was related to xpmem, it wasn't.  My first iterations of the docker image were missing pkg-config, and that was the reason the ugni configure step failed. I found the output file of a failed run:

```
configure: WARNING: KNEM requested but required file (knem_io.h) could not be found
configure: XPMEM - failed to open the requested location (guess), guessing ...
/__w/1/s/contrib/../configure: line 25057: pkg-config: command not found
checking cray-ugni... /__w/1/s/contrib/../configure: line 25191: pkg-config: command not found
```
I fixed the pkg-config issue when making the build_release_pgk() step work later